.drone.yml

```diff
@@ -144,10 +144,24 @@ steps:
       IGNORE_UNFIXED: true
       ALLOW_CVE_LIST_FILE: hof-services-config/Change_Of_Address/trivy-cve-exceptions.txt
     when:
-      event:
-      - pull_request
-      - push
-      - tag
+      event: [push, pull_request, tag]
+
+  # Trivy Security Scannner for scanning OS related vulnerabilities in Base image of Dockerfile
+  - name: scan_image_os
+    pull: always
+    image: 340268328991.dkr.ecr.eu-west-2.amazonaws.com/acp/trivy/client:latest
+    resources:
+      limits:
+        cpu: 1000
+        memory: 1024Mi
+    environment:
+      IMAGE_NAME: node:18-alpine@sha256:2322b1bb3917b313f2e9308395aa5c39d51b91cc92a5d4d5be6d0451fcfb4d24
```

Comment:
Shouldn't we test `coa:${DRONE_COMMIT_SHA}` to make sure we are testing the final product, in case the node version is updated for the application at some point but was overlooked and not updated at that step?
I don't see the point of testing node:18-alpine with the same SHA over and over again?

And if so then steps `scan_image_os` and `scan_node_packages` are essentially the same?

Reply:
The main issue our team has noticed is Current Trivy Scan step is scanning only the YARN/npm node packages in the coa image skipping the vulnerabilities in the base image used by coa image.
We have tested by scanning the base image used by the App image in ARE as requested by Team and Delivery Managers. This helped us in detecting hidden vulnerabilities in OS
https://drone-gh.acp.homeoffice.gov.uk/UKHomeOffice/AppealRightsExhausted/1013/1/8